### PR TITLE
Ansible changes for sFlow max header size

### PR DESCRIPTION
**Pull request by aravindmani-1**
##### SUMMARY
Ansible implementation for Sflow max header size 
##### GitHub Issues
N/A

| GitHub Issue # |
| -------------- |
| |

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
sonic_sflow

##### OUTPUT
**Ansible report:** 
[regression-2024-11-20-14-42-18.pdf](https://github.com/user-attachments/files/17828005/regression-2024-11-20-14-42-18.pdf)

**Code coverage UT:**
[code coverage UT.txt](https://github.com/user-attachments/files/17783985/code.coverage.UT.txt)

##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

##### How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: changelogs/fragments/419-sflow-max-header-size.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - sonic_ldap - Add max header size support in sonic_sflow module(https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/419)
```
Comment (awhaley-dell) [inline changelogs/fragments/419-sflow-max-header-size.yaml:2]: Please add a space between the word "module" and the "(" character.
Add a period after the ")" character as well.
The fix should look like this: 
- sonic_ldap - Add max header size support in sonic_sflow module (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/419).

File: tests/regression/roles/sonic_sflow/defaults/main.yml [hunk 2]
```diff
@@ -3,11 +3,12 @@ module_name: sonic_sflow
 
 tests:
   - name: "test_case_01_merged_globals"
-    description: testing setting merged, polling interval, agent name, and global sampling rate
+    description: testing setting merged, polling interval,max header size, agent name, and global sampling rate
```
Comment (awhaley-dell) [inline tests/regression/roles/sonic_sflow/defaults/main.yml:6]: Add a space before the word "max" on this line please.
The fix should look like this:
description: testing setting merged, polling interval, max header size, agent name, and global sampling rate

**Review by awhaley-dell — CHANGES_REQUESTED** [review]
Overall, the code looks good.
Only feedback requests are minor formatting and punctuation errors that should be corrected.

File: changelogs/fragments/419-sflow-max-header-size.yaml (hunk 1 above)
Comment (awhaley-dell) [inline changelogs/fragments/419-sflow-max-header-size.yaml:2]:
```suggestion
  - sonic_ldap - Add max header size support in sonic_sflow module (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/419).
```

File: tests/regression/roles/sonic_sflow/defaults/main.yml (hunk 2 above)
Comment (awhaley-dell) [inline tests/regression/roles/sonic_sflow/defaults/main.yml:6]:
```suggestion
    description: testing setting merged, polling interval, max header size, agent name, and global sampling rate
```

File: changelogs/fragments/419-sflow-max-header-size.yaml (hunk 1 above)
Comment (aravindmani-1) [reply changelogs/fragments/419-sflow-max-header-size.yaml]: Accepted. committed new changes.

File: tests/regression/roles/sonic_sflow/defaults/main.yml (hunk 2 above)
Comment (aravindmani-1) [reply tests/regression/roles/sonic_sflow/defaults/main.yml]: Accepted. committed new changes.

Comment (aravindmani-1) [reply tests/regression/roles/sonic_sflow/defaults/main.yml]: Done.

**Review by awhaley-dell — APPROVED** [review]
Looks good now. Approving. Thanks for making the requested changes.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
I am posting one comment for a code line that looks incorrect.

- I also would like to request further investigation into the regression test output for "test_case_07_overridden.1". There are two interfaces specified in the "input", but neither of these interfaces is present in the "After" display. This indicates a bug that was apparently already present in this resource module, but please try to determine why this is happening. If possible, please update the change set for this PR with code changes to address this problem. 
- The current test results have some missing text in the rightmost column displayed, and the "real" rightmost column from the html output is completely missing. Please reformat this printed result to print in "landscape" mode, scaled as needed to fit on the page that is being printed.

File: plugins/module_utils/network/sonic/config/sflow/sflow.py [hunk 3]
```diff
@@ -315,6 +319,9 @@ def validate_normalize_config(self, config):
         if config is not None and config.get("polling_interval") is not None:
             if not (int(config["polling_interval"]) == 0 or int(config["polling_interval"]) in range(5, 301)):
                 self._module.fail_json(msg="polling interval out of range. must be 0 or in the range 5-300 inclusive", code=1)
+        if config is not None and config.get("max_header_size") is not None:
+            if not (int(config["max_header_size"] % 128) == 0 or int(config["max_header_size"]) in range(128, 1024)):
```
[the remark this answers is not in the document]
Comment (aravindmani-1) [reply plugins/module_utils/network/sonic/config/sflow/sflow.py]: Thanks Kerry. Have addressed the suggested changes in latest commit and attached the regression results in Description.

**Review by kerry-meyer — APPROVED** [review]
Thank you for making the additional requested change and for re-posting the test results. All current proposed changes and test results look good.

Approved.

**Pull request merged**